sdks/python/apache_beam/io/gcp/bigtable_io_test.py

```diff
@@ -0,0 +1,172 @@
+  instance_id = DEFAULT_TABLE_PREFIX + "-" + str(uuid.uuid4())[:8]
+  cluster_id = DEFAULT_TABLE_PREFIX + "-" + str(uuid.uuid4())[:8]
+  table_id = DEFAULT_TABLE_PREFIX + "-" + str(uuid.uuid4())[:8]
+  number = 500
+  LOCATION_ID = "us-east1-b"
+
+  def setUp(self):
+    try:
+      from google.cloud.bigtable import enums
+      self.STORAGE_TYPE = enums.StorageType.HDD
+      self.INSTANCE_TYPE = enums.Instance.Type.DEVELOPMENT
+    except ImportError:
+      self.STORAGE_TYPE = 2
+      self.INSTANCE_TYPE = 2
+
+    self.test_pipeline = TestPipeline(is_integration_test=True)
+    self.runner_name = type(self.test_pipeline.runner).__name__
+    self.project = self.test_pipeline.get_option('project')
+    self.client = Client(project=self.project, admin=True)
+    self._create_instance_table()
+
+  def tearDown(self):
+    if self.instance.exists():
+      self.instance.delete()
+
+  def test_bigtable_write_python(self):
+    number = self.number
+    config = BigtableWriteConfiguration(self.project, self.instance_id,
+                                        self.table_id)
+    pipeline_args = self.test_pipeline.options_list
+    pipeline_options = PipelineOptions(pipeline_args)
+    rows = self._generate_mutation_data(number)
+
+    with beam.Pipeline(options=pipeline_options) as pipeline:
+      _ = (
+          pipeline
+          | 'Generate Row Values' >> beam.Create(rows)
+          | 'Generate Direct Rows' >> beam.ParDo(GenerateDirectRows())
+          | 'Write to BT' >> beam.ParDo(WriteToBigtable(config)))
+
+      result = pipeline.run()
+      result.wait_until_finish()
+
+      assert result.state == PipelineState.DONE
+      assert self._check_table(number)
+      if not hasattr(result, 'has_job') or result.has_job:
+        read_filter = MetricsFilter().with_name('Written Row')
+        query_result = result.metrics().query(read_filter)
+        if query_result['counters']:
+          read_counter = query_result['counters'][0]
+
+          logging.info('Number of Rows: %d', read_counter.committed)
+          assert read_counter.committed == number
+
+  def _create_instance_table(self):
+    """ Prepare all necesary for the test
+    """
+    self.instance = self.client.instance(self.instance_id,
+                                         instance_type=self.INSTANCE_TYPE)
+
+    if not self.instance.exists():
+      cluster = self.instance.cluster(self.cluster_id,
+                                      self.LOCATION_ID,
+                                      default_storage_type=self.STORAGE_TYPE)
+      self.instance.create(clusters=[cluster])
+    self.table = self.instance.table(self.table_id)
+
+    if not self.table.exists():
+      max_versions_rule = column_family.MaxVersionsGCRule(2)
+      column_family_id = 'cf1'
+      column_families = {column_family_id: max_versions_rule}
+      self.table.create(column_families=column_families)
+
+  def _check_table(self, number):
+    return len(list(self.table.read_rows())) == number
```

Comment:
This looks elegant.  However, it seems space intensive.  something like `sum(1 for _ in iter)`

Also please add a filter that only returns 1 column, with an empty value.